starfish/util/logging.py

```diff
@@ -0,0 +1,40 @@
+import platform
+import subprocess
+from functools import lru_cache
+from json import JSONEncoder
+
+
+from starfish.types import CORE_DEPENDENCIES
+
+
+@lru_cache(maxsize=1)
+def get_core_dependency_info():
+    dependency_info = dict()
+    for dependency in CORE_DEPENDENCIES:
+        ps = subprocess.Popen(('pip', 'show', dependency), stdout=subprocess.PIPE)
+        version = subprocess.check_output(('grep', 'Version'), stdin=ps.stdout).strip()
+        dependency_info[dependency] = version
+    return dependency_info
+
+
+@lru_cache(maxsize=1)
+def get_git_commit_hash():
+    return subprocess.check_output(["git", "describe", "--always"]).strip()
```

Comment:
This will fail if either or both of `starfish was installed via pip` or `user is not int the starfish directory` hold true

Reply:
That is an excellent point